lib/new-admin/graphql/schema.js

```diff
@@ -402,7 +409,8 @@ const resolvers = {
     resetAccounts: (...[, { schemaVersion }]) => settingsLoader.resetAccounts(schemaVersion),
     setCustomer: (root, args, context, info) => {
       const token = context.req.cookies && context.req.cookies.token
-      return customers.updateCustomer(args.customerId, args.customerInput, token)
+      if (args.customerId !== anonymous.uuid) return customers.updateCustomer(args.customerId, args.customerInput, token)
+      return customers.getCustomerById(args.customerId)
```

Comment:
It's cleaner if you do the other way around

```suggestion
      if (args.customerId === anonymous.uuid) return customers.getCustomerById(args.customerId)
      return customers.updateCustomer(args.customerId, args.customerInput, token)
      
```